src/rust/engine/src/nodes.rs

```diff
@@ -883,7 +886,7 @@ pub struct Snapshot {
 }
 
 impl Snapshot {
-  fn create(context: Context, path_globs: PathGlobs) -> NodeFuture<fs::Snapshot> {
+  fn create(context: Arc<Context>, path_globs: PathGlobs) -> NodeFuture<fs::Snapshot> {
```

Comment:
`Context` implements `Clone` (cheaply): no need to wrap it in an `Arc`.

Reply:
It looks like rather than taking an `Arc<GFD>`, `Snapshot::from_path_stats` should maybe take a `GFD` with a `Clone` bound?:
```
-  pub fn from_path_stats<GFD: GetFileDigest<Error> + Sized, Error: fmt::Debug + 'static + Send>(
+  pub fn from_path_stats<GFD: GetFileDigest<Error> + Clone, Error: fmt::Debug + 'static + Send>(
     store: Arc<Store>,
-    file_digester: Arc<GFD>,
+    file_digester: GFD,
     mut path_stats: Vec<PathStat>,
   ) -> BoxFuture<Snapshot, String> {
```

Reply:
Done